doc/user/content/sql/explain-timestamp.md

```diff
@@ -0,0 +1,192 @@
+---
+title: "EXPLAIN TIMESTAMP"
+description: "`EXPLAIN TIMESTAMP` displays the timestamps used for a `SELECT` statement."
+menu:
+  main:
+    parent: commands
+---
+
+`EXPLAIN TIMESTAMP` displays the timestamps used for a `SELECT` statement -- valuable information to acknowledge query delays.
+
+{{< note >}}
+The `EXPLAIN PLAN` documentation now lives in [its own page](/sql/explain-plan).
+{{</ note >}}
+
+{{< warning >}}
+`EXPLAIN` is not part of Materialize's stable interface and is not subject to
+our backwards compatibility guarantee. The syntax and output of `EXPLAIN` may
+change arbitrarily in future versions of Materialize.
+{{< /warning >}}
+
+## Syntax
+
+{{< diagram "explain-timestamp.svg" >}}
+
+### Output format
+
+You can select between `JSON` and `TEXT` for the output format of `EXPLAIN TIMESTAMP`. Non-text
+output is more machine-readable and can be parsed by common graph visualization libraries,
+while formatted text is more human-readable.
+
+Output type | Description
+------|-----
+**TEXT** | Format the explanation output as UTF-8 text.
+**JSON** | Format the explanation output as a JSON object.
+
+## Details
+
+The explanation is divided in two parts:
+
+1. Determinations for a timestamp
+2. Sources frontiers
+
+Having a _query timestamp_ outside the _[read, write)_ frontier values of a source can explain the presence of delays. While in the middle, the space of processed but not yet compacted data, allows building and returning a correct result immediately.
+
+### Determinations for a timestamp
+
+Queries in Materialize have a logical timestamp, known as _query timestamp_. It plays a critical role to return a correct result. Returning a correct result implies retrieving data with the same logical time from each source present in a query.
+
+In this case, sources are objects providing data: materialized views, views, indexes, tables, or sources. Each will have a pair of logical timestamps frontiers, denoted as _sources frontiers_.
+
+This section contains the following fields:
+
+Field | Meaning | Example
+---------|---------|---------
+**query timestamp** | The query timestamp value |`1673612424151 (2023-01-13 12:20:24.151)`
+**oracle read** | The value of the timeline's oracle timestamp, if used. | `1673612424151 (2023-01-13 12:20:24.151)`
+**largest not in advance of upper** | The largest timestamp not in advance of upper. | `1673612424151 (2023-01-13 12:20:24.151)`
+**since** | The maximum read frontier of all involved sources. | `[1673612423000 (2023-01-13 12:20:23.000)]`
+**upper** | The minimum write frontier of all involved sources | `[1673612424152 (2023-01-13 12:20:24.152)]`
+**can respond immediately** | Returns true when the **query timestamp** is greater or equal to **since** and lower than **upper** | `true`
+**timeline** | The type of timeline the query's timestamp belongs | `Some(EpochMilliseconds)`
+
+Possible timeline values are:
```

Comment:
Users have no way of making non-epoch timelines, so the following table and this sentence might be more confusing than they are worth. It might be notable to say that `None` there means the query is known to be constant across all timestamps.